radicle-httpd/src/api.rs

```diff
@@ -51,15 +51,40 @@ impl Context {
         let doc = repo.identity_of(self.profile.id())?;
         let payload = doc.project()?;
         let delegates = doc.delegates;
-        let issues = (Issues::open(self.profile.public_key, &repo)?).count()?;
-        let patches = (Patches::open(self.profile.public_key, &repo)?).count()?;
+        let issues = Issues::open(self.profile.public_key, &repo)?.all()?;
+        let (open_issues, closed_issues): (Vec<_>, Vec<_>) = issues
+            .filter_map(|s| s.ok())
+            .partition(|(_, issue, _)| issue.state() == &issue::State::Open);
+        let patches = Patches::open(self.profile.public_key, &repo)?.all()?;
+        let patches: Vec<_> = patches.filter_map(|s| s.ok()).collect();
+        let draft_patches: Vec<_> = patches
+            .clone()
+            .into_iter()
+            .filter(|(_, patch, _)| patch.state() == patch::State::Draft)
+            .collect();
+        let proposed_patches: Vec<_> = patches
+            .clone()
+            .into_iter()
+            .filter(|(_, patch, _)| patch.state() == patch::State::Proposed)
+            .collect();
+        let archived_patches: Vec<_> = patches
+            .into_iter()
+            .filter(|(_, patch, _)| patch.state() == patch::State::Archived)
+            .collect();
```

Comment:
Can do the following (and the same for issues)
```suggestion
        let patch_stats = patches
            .into_iter()
            .fold(PatchState::default(), |mut state, p| {
                match p.state() {
                    patch::State::Draft => state.draft += 1,
                    patch::State::Proposed => state.proposed += 1,
                    patch::State::Archived => state.archived += 1,
                }
                state
            });
```

Reply:
TIL about `fold` very nice!
Addressed here 3683cdd

Reply:
Nice :) The fold logic is duplicated across the 3 handlers, hence why I think there could a method to de-duplicate it and just call the method in the 3 places.